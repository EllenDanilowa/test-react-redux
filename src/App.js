import React, { Component } from 'react';
import {
  BrowserRouter as Router,
  Switch,
  Route,
  Link
} from 'react-router-dom';
import AllMerchantsPage from './routes/AllMerchantsPage/AllMerchantsPage'; // think about naming
import NewMerchantPage from './routes/NewMerchantPage/NewMerchantPage';
import logo from './logo.png';
import './App.css';

class App extends Component {
  render() {
    return (
      <Router>
        <div className="App">
          <div className="App-header">
            <Link to="/">
              <img src={logo} className="App-logo" alt="logo" />
            </Link>
          </div>
          <main>
            <Switch>
              <Route path="/add">
                <NewMerchantPage />
              </Route>
              {/* <Route path="/edit">
                <Edit />
              </Route> */}
              <Route path="/">
                <AllMerchantsPage />
              </Route>
            </Switch>
          </main>
        </div>
      </Router>
    );
  }
}

export default App;