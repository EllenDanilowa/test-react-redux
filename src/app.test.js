import React from 'react';
import renderer from 'react-test-renderer';
import App from './app';

describe('App', () => {
  let element;

  const createElement = () => {
    const template = (<App/>);

    return renderer.create(template).toJSON();
  };

  it('renders component correctly', () => {
    element = createElement();

    expect(element).toMatchSnapshot();
  });
});
