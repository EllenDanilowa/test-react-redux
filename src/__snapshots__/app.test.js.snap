// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App renders component correctly 1`] = `
Array [
  <header
    className="sc-fzoxKX hyEIdl"
>
    <a
        href="/"
        onClick={[Function]}
    >
        <img
            alt="logo"
            className="sc-fzqPZZ kcCvUZ"
            src="logo.png"
        />
    </a>
</header>,
  <main>
    <div
        className="sc-fzoXzr hNPPhd"
    >
        <div
            className="sc-fzpjYC boDesf"
        >
            <h3
                className="sc-fzoLag kmISQr"
            >
                All merchants
            </h3>
            <div>
                <a
                    href="/create"
                    onClick={[Function]}
                >
                    <img
                        className="sc-fznxsB wWyiq"
                        src="plus.svg"
                    />
                    Add new merchant
                </a>
            </div>
        </div>
        <div />
    </div>
</main>,
]
`;
